libraries/AP_EFI/AP_EFI.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_EFI.h"
+
+#if EFI_ENABLED
+
+#include "AP_EFI_Serial_MS.h"
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL& hal;
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_EFI::var_info[] = {
+    // @Param: _TYPE
+    // @DisplayName: EFI communication type
+    // @Description: What method of communication is used for EFI #1
+    // @Values: 0:None,1:Serial-MS
+    // @User: Advanced
+    // @RebootRequired: True
+    AP_GROUPINFO_FLAGS("_TYPE", 1, AP_EFI, type, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: _COEF1
+    // @DisplayName: EFI Calibration Coefficient 1
+    // @Description: Used to calibrate fuel flow for MS protocol (Slope)
+    // @Range: 0 1
+    // @User: Advanced
+    // @RebootRequired: False
+    AP_GROUPINFO("_COEF1", 2, AP_EFI, coef1, 0),
+
+    // @Param: _COEF2
+    // @DisplayName: EFI Calibration Coefficient 2
+    // @Description: Used to calibrate fuel flow for MS protocol (Offset)
+    // @Values: 0 10
```

Comment:
This should have been a range, not values.

Reply:
fixed